contracts/SpecificGroupStrategy.sol

```diff
@@ -337,53 +373,133 @@ contract SpecificGroupStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeab
      * @notice Used to withdraw CELO from the system from specific group strategy
      * that account voted for previously. It is expected that strategy will be balanced.
      * For balancing use `rebalance` function
-     * @param group The validator group that we want to withdraw from.
-     * @param celoWithdrawalAmount The amount of CELO to withdraw.
-     * @param stCeloWithdrawalAmount The amount of stCELO to withdraw.
+     * @param strategy The validator group that we want to withdraw from.
+     * @param celoWithdrawalAmount The amount of stCELO to withdraw.
      * @return groups The groups to withdraw from.
      * @return votes The amount to withdraw from each group.
      */
     function calculateAndUpdateForWithdrawalTransfer(
-        address group,
+        address strategy,
         uint256 celoWithdrawalAmount,
         uint256 stCeloWithdrawalAmount
     ) public onlyManager returns (address[] memory groups, uint256[] memory votes) {
         if (specificGroupStrategies.length() == 0) {
             revert NoGroups();
         }
 
-        groups = new address[](1);
-        votes = new uint256[](1);
-        groups[0] = group;
-        votes[0] = celoWithdrawalAmount;
+        if (stCeloWithdrawalAmount > stCeloInStrategy[strategy]) {
+            revert CantWithdrawAccordingToStrategy(strategy);
+        }
 
-        if (stCeloWithdrawalAmount > stCeloInGroup[group]) {
-            revert CantWithdrawAccordingToStrategy(group);
+        uint256 overflowingStCelo = stCeloInStrategyOverflown[strategy];
+        if (overflowingStCelo > 0) {
+            uint256 overflowingCelo = IManager(manager).toCelo(overflowingStCelo);
```

Comment:
One thing to think about here: if we knew the target strategy here, we could short-circuit if the target strategy was the default, since we don't actually have to withdraw anything from it, just need to update the amount overflowed for this specific strategy.

Reply:
This will actually happen on Account level. When there will be some votes to be scheduled and some votes to be revoked for same validator group, it will just cancel itself out.

Since I didn't consider this scenario too common it felt like having this mechanism on Account level is sufficient. User will have to pay a bit more for gas fees but protocol will not be hurt in anyway.